std/path.d

```diff
@@ -691,9 +691,21 @@ if (isRandomAccessRange!R && hasSlicing!R && hasLength!R && isSomeChar!(ElementT
         A slice of $(D path).
 */
 auto rootName(R)(R path)
-if ((isRandomAccessRange!R && hasSlicing!R && hasLength!R && isSomeChar!(ElementType!R) ||
-    isNarrowString!R) &&
-    !isConvertibleToString!R)
+if (isRandomAccessRange!R && hasSlicing!R && hasLength!R && isSomeChar!(ElementType!R) && !isSomeString!R)
+{
+    return _rootName(path);
+}
+
+/// ditto
+auto rootName(C)(C[] path)
+if (isSomeChar!C)
```

Comment:
They aren't redundant. That's the point. The only `@safe` option with anything using `isConvertibleToString` is to create an overload that explicitly takes a dynamic array of a characters. And it's impossible to deprecate any part of that and have it be `@safe` in the interim. Having multiple functions like this is what's required to actually safely support accepting anything that implicitly converts anything to a string. The conversion _must_ happen at the call site, or you're screwed, and that won't work with template constraints that accept the whole type.